service/src/main/java/de/adorsys/psd2/sandbox/xs2a/service/redirect/RedirectService.java

```diff
@@ -0,0 +1,113 @@
+import de.adorsys.psd2.consent.repository.AisConsentAuthorizationRepository;
+import de.adorsys.psd2.consent.repository.AisConsentRepository;
+import de.adorsys.psd2.consent.repository.PisAuthorizationRepository;
+import de.adorsys.psd2.consent.repository.PisPaymentDataRepository;
+import de.adorsys.psd2.xs2a.core.consent.ConsentStatus;
+import de.adorsys.psd2.xs2a.core.pis.TransactionStatus;
+import de.adorsys.psd2.xs2a.core.sca.ScaStatus;
+import java.util.List;
+import java.util.Optional;
+import javax.transaction.Transactional;
+import org.springframework.stereotype.Service;
+
+@Service
+@Transactional
+public class RedirectService {
+
+  private PisPaymentDataRepository pisPaymentDataRepository;
+  private PisAuthorizationRepository pisAuthorizationRepository;
+  private AisConsentAuthorizationRepository aisConsentAuthorizationRepository;
+  private AisConsentRepository aisConsentRepository;
+
+  /**
+   * Create a new RedirectService instance.
+   *
+   * @param pisPaymentDataRepository          PisPaymentRepository
+   * @param pisAuthorizationRepository        PisAuthorizationRepository
+   * @param aisConsentAuthorizationRepository AisConsentAuthorizationRepository
+   * @param aisConsentRepository              AisConsentRepository
+   */
+  public RedirectService(PisPaymentDataRepository pisPaymentDataRepository,
+      PisAuthorizationRepository pisAuthorizationRepository,
+      AisConsentAuthorizationRepository aisConsentAuthorizationRepository,
+      AisConsentRepository aisConsentRepository) {
+    this.pisPaymentDataRepository = pisPaymentDataRepository;
+    this.pisAuthorizationRepository = pisAuthorizationRepository;
+    this.aisConsentAuthorizationRepository = aisConsentAuthorizationRepository;
+    this.aisConsentRepository = aisConsentRepository;
+  }
+
+  /**
+   * Sets consent status in CMS depending on PSU-ID.
+   *
+   * @param externalId Consent Id
+   * @param psuId      Psu Id
+   */
+  public void handleConsentCreationRedirectRequest(String externalId, String psuId) {
+    Optional<AisConsentAuthorization> aisConsentAuthorization = aisConsentAuthorizationRepository
+        .findByExternalId(externalId);
+
+    if (!aisConsentAuthorization.isPresent()) {
+      //TODO handle error case
+      return;
+    }
+    AisConsentAuthorization aisConsentAuth = aisConsentAuthorization.get();
+
+    AisConsent consent = aisConsentRepository.findOne(aisConsentAuth.getId());
+    // TODO: Set status of PSU from TestDataService
+    consent.setConsentStatus(ConsentStatus.VALID);
+    aisConsentAuth.setScaStatus(ScaStatus.PSUAUTHENTICATED);
+
+    aisConsentRepository.save(consent);
+    aisConsentAuthorizationRepository.save(aisConsentAuth);
+  }
+
+  /**
+   * Sets payment status in CMS depending on PSU-ID.
+   *
+   * @param externalId Payment Id
+   * @param psuId      Psu Id
+   */
+  public void handlePaymentRedirectRequest(String externalId, String psuId, boolean isInit) {
```

Comment:
`isInit` is unused. Why do we need different redirects for cancellation and initiation? How would we handle this in the banking UI?

Reply:
We will need `isInit` to set different status for initiation and cancellation flows.
- Successfull init response results in ACCP
- Successfull cancel response results in CANC

And this is the same procedure in the banking Ui.

I added a TODO for this parameter